CMakeLists.txt

```diff
@@ -15,146 +14,169 @@ endif()
-  SET(PRE_FLAG  KOKKOS_CFG_DAG_${PRECURSOR})
-  SET(POST_FLAG KOKKOS_CFG_DAG_${SUCCESSOR})
-  IF (NOT ${PRE_FLAG})
-    MESSAGE(FATAL_ERROR "Bad CMake refactor: feature ${SUCCESSOR} cannot be configured until ${PRECURSOR} is configured")
-  ENDIF()
-  GLOBAL_SET(${POST_FLAG} On)
-ENDFUNCTION()
-
-
-LIST(APPEND CMAKE_MODULE_PATH cmake/Modules)
-
-IF(NOT KOKKOS_HAS_TRILINOS)
-  set(CMAKE_DISABLE_SOURCE_CHANGES ON)
-  set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
-
-  # What language are we compiling Kokkos as
-  # downstream dependencies need to match this!
-  SET(KOKKOS_COMPILE_LANGUAGE CXX)
-  # use lower case here since we didn't parse options yet
-  IF (Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE AND Kokkos_ENABLE_CUDA)
-
-    # Without this as a language for the package we would get a C++ compiler enabled.
-    # but we still need a C++ compiler even if we build all our cpp files as CUDA only
-    # because otherwise the C++ features don't work etc.
-    # This is just the rather odd way CMake does this, since CUDA doesn't imply C++ even
-    # though it is a C++ extension ... (but I guess it didn't use to be back in CUDA 4 or 5
-    # days.
-    SET(KOKKOS_INTERNAL_EXTRA_COMPILE_LANGUAGE CXX)
-
-    SET(KOKKOS_COMPILE_LANGUAGE CUDA)
-  ENDIF()
-  # use lower case here since we haven't parsed options yet
-  IF (Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE AND Kokkos_ENABLE_HIP)
-
-    # Without this as a language for the package we would get a C++ compiler enabled.
-    # but we still need a C++ compiler even if we build all our cpp files as HIP only
-    # because otherwise the C++ features don't work etc.
-    SET(KOKKOS_INTERNAL_EXTRA_COMPILE_LANGUAGE CXX)
-
-    SET(KOKKOS_COMPILE_LANGUAGE HIP)
-  ENDIF()
-
-  IF (Spack_WORKAROUND)
-    IF (Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
-      MESSAGE(FATAL_ERROR "Can't currently use Kokkos_ENABLE_COMPILER_AS_CMAKE_LANGUAGE in a spack installation!")
-    ENDIF()
-
-    #if we are explicitly using Spack for development,
-    #nuke the Spack compiler
-    SET(SPACK_CXX $ENV{SPACK_CXX})
-    IF(SPACK_CXX)
-      SET(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
-      SET(ENV{CXX} ${SPACK_CXX})
-    ENDIF()
-  ENDIF()
-  # Always call the project command to define Kokkos_ variables
-  # and to make sure that C++ is an enabled language
-  PROJECT(Kokkos ${KOKKOS_COMPILE_LANGUAGE} ${KOKKOS_INTERNAL_EXTRA_COMPILE_LANGUAGE})
-  IF(NOT HAS_PARENT)
-    IF (NOT CMAKE_BUILD_TYPE)
-      SET(DEFAULT_BUILD_TYPE "RelWithDebInfo")
-      MESSAGE(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' as none was specified.")
-      SET(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE STRING
+set(KOKKOS_CFG_DAG_NONE On) #sentinel to indicate no dependencies
+set(KOKKOS_CFG_DAG_DEVICES_DONE Off)
+set(KOKKOS_CFG_DAG_OPTIONS_DONE Off)
+set(KOKKOS_CFG_DAG_ARCH_DONE Off)
+set(KOKKOS_CFG_DAG_CXX_STD_DONE Off)
+set(KOKKOS_CFG_DAG_COMPILER_ID_DONE Off)
+function(KOKKOS_CFG_DEPENDS SUCCESSOR PRECURSOR)
```

Comment:
I'm not sure that modifying the whole `CMake` base for case is a good idea.

You'll make *git* history much harder when we need to go back to understand things.

I would strongly suggest to run your format check on the new code only.

Reply:
> You'll make _git_ history much harder when we need to go back to understand things.

There's a way to avoid it by using `.git-blame-ignore-revs`.

Reply:
Yep, but frankly it's just an additional burden. But I will not block, mess with it if you deem it necessary.

Reply:
@romintomasetti Looks like you are currently blocking.

Reply:
Oh sorry about that !